src/backend/libs/efiling-lookup-client/src/test/java/ca/bc/gov/open/jag/efilinglookupclient/CSOLookupServiceImplTest.java

```diff
@@ -3,54 +3,113 @@
+    public static final String ENT_USER_ID = "entUserId";
+    public static final String SERVICE_TYPE_CD = "serviceTypeCd";
+    public static final String UPD_USER_ID = "updUserId";
+    public static final String SERVICE_ID_EXCEPTION = "exception";
 
-    @InjectMocks
     CSOLookupServiceImpl sut;
 
     @Mock
-    LookupFacadeBean mocklookupFacadeItf;
+    LookupFacadeBean lookupFacadeBeanMock;
 
     @Mock
     ServiceFee serviceFeeMock;
+    @Mock
+    private XMLGregorianCalendar updDtmMock;
+    @Mock
+    private XMLGregorianCalendar expiryDtMock;
+    @Mock
+    private XMLGregorianCalendar entDtmMock;
+    @Mock
+    private XMLGregorianCalendar effectiveDtMock;
+
 
     @BeforeEach
-    public void init() throws NestedEjbException_Exception {
+    public void init() throws NestedEjbException_Exception, DatatypeConfigurationException {
 
         MockitoAnnotations.initMocks(this);
-        Mockito.when(serviceFeeMock.getFeeAmt()).thenReturn(BigDecimal.valueOf(0));
-        Mockito.when(mocklookupFacadeItf.getServiceFee(any(), any())).thenReturn(serviceFeeMock);
+        Mockito.when(serviceFeeMock.getFeeAmt()).thenReturn(BigDecimal.TEN);
+        Mockito.when(serviceFeeMock.getEntUserId()).thenReturn(ENT_USER_ID);
+        Mockito.when(serviceFeeMock.getServiceTypeCd()).thenReturn(SERVICE_TYPE_CD);
+        Mockito.when(serviceFeeMock.getUpdUserId()).thenReturn(UPD_USER_ID);
+        Mockito.when(updDtmMock.toGregorianCalendar()).thenReturn(new DateTime(2020, 7, 1, 10,10).toGregorianCalendar());
+        Mockito.when(expiryDtMock.toGregorianCalendar()).thenReturn(new DateTime(2020, 7, 2, 10,10).toGregorianCalendar());
+        Mockito.when(entDtmMock.toGregorianCalendar()).thenReturn(new DateTime(2020, 7, 3, 10,10).toGregorianCalendar());
+        Mockito.when(effectiveDtMock.toGregorianCalendar()).thenReturn(new DateTime(2020, 7, 4, 10,10).toGregorianCalendar());
+        Mockito
+                .when(serviceFeeMock.getEffectiveDt())
+                .thenReturn(effectiveDtMock);
+        Mockito
+                .when(serviceFeeMock.getEntDtm())
+                .thenReturn(entDtmMock);
+        Mockito
+                .when(serviceFeeMock.getExpiryDt())
+                .thenReturn(expiryDtMock);
+        Mockito
+                .when(serviceFeeMock.getUpdDtm())
+                .thenReturn(updDtmMock);
+
+        Mockito
+                .when(lookupFacadeBeanMock.getServiceFee(Mockito.eq(SERVICE_ID), any()))
+                .thenReturn(serviceFeeMock);
+
+        Mockito
+                .when(lookupFacadeBeanMock.getServiceFee(Mockito.eq(SERVICE_ID_EXCEPTION), Mockito.any()))
+                .thenThrow(new NestedEjbException_Exception("random"));
+
+        sut = new CSOLookupServiceImpl(lookupFacadeBeanMock);
+
     }
 
-    @DisplayName("CASE 1: getServiceFee called with empty serviceId")
+    @DisplayName("Exception: with null service id should throw IllegalArgumentException")
```

Comment:
is there a status code associated with this?

Reply:
nope